common/src/main/java/tachyon/network/protocol/databuffer/DataBuffer.java

```diff
@@ -42,4 +42,11 @@
    * @return a read-only ByteBuffer representing the data
    */
   public abstract ByteBuffer getReadOnlyByteBuffer();
+
+  /**
+  * Release the underlying buffer of this DataBuffer if no longer needed.
+  *
+  * @return True if the underlying buffer is released or no need to release
+  */
+  public abstract boolean release();
```

Comment:
Suppose a unit test on `DataNettyBuffer`, which means test the functionality without the `MessageToMessageDecoder`. It might look like:
```
ByteBuf buf = Unpooled.buffer(128); //counter = 1
DataBuffer dataBuffer = new DataNettyBuffer(buf); //counter = 2
databuffer.release(); // return false
```